### PR TITLE
Added config and backward compatability

### PR DESCRIPTION
**Pull request by applejag**
## Summary

- Added wharf-core/pkg/config loading
- Changed to use Config throughout instead of env vars
- Added backward compatability with previous environment variables

## Motivation

Unified and more power configuration system

## Unresolved questions

> @fredx30
> I expected a default config file to be committed to the repo with the defaults set. Its much simpler to change a config than to invent one that conforms to a set standard where description is scarce.
>
> https://github.com/iver-wharf/wharf-api/pull/38#pullrequestreview-703929360

Please continue that discussion over at the wharf-api PR linked above.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,22 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
...
+- Added config loading from YAML files using
+  `github.com/iver-wharf/wharf-core/pkg/config` together with new config models
+  for configuring wharf-provider-github. See `config.go` or the reference
+  documentation on the `Config` type for information on how to configure
+  wharf-provider-github. (#19)
+
+- Added config for setting bind address and port. (#14, #19)
+
+  - Environment variable: `WHARF_API_URL`
+  - YAML: `api.url`
```
Comment (Alexamakans) [inline CHANGELOG.md]: This doesn't seem right.
Shouldn't it be `WHARF_HTTP_BINDADDRESS` and `http.bindAddress`?

**Review by Alexamakans — CHANGES_REQUESTED** [review]
I didn't find anything else that seemed weird 👍🏻

File: CHANGELOG.md (hunk 1 above)
Comment (applejag) [reply CHANGELOG.md]: Yea, probably should

**Review by Alexamakans — APPROVED** [review]

**Comment by Alexamakans** [comment]
Noticed you haven't requested review yet, oops.
Apologies for the premature review.

**Comment by applejag** [comment]
> Noticed you haven't requested review yet, oops.
> Apologies for the premature review.

No worries. If you review while it's a draft then there's a high chance you'll have to review again once it has been marked as ready. For me as the PR author it's the more reviews the better.

**Comment by Alexamakans** [comment]
> > Noticed you haven't requested review yet, oops.
> > Apologies for the premature review.
> 
> No worries. If you review while it's a draft then there's a high chance you'll have to review again once it has been marked as ready. For me as the PR author it's the more reviews the better.

Coolio :+1:

**Review by Alexamakans — APPROVED** [review]

**Review by fredx30 — APPROVED** [review]

**Pull request merged**